packages/core/src/electron-main/electron-application.ts

```diff
@@ -0,0 +1,440 @@
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable, named } from 'inversify';
+import { session, screen, globalShortcut, app, BrowserWindow, BrowserWindowConstructorOptions, Event as ElectronEvent, dialog } from 'electron';
+import * as path from 'path';
+import { Argv } from 'yargs';
+import { AddressInfo } from 'net';
+import { promises as fs } from 'fs';
+import { fork, ForkOptions } from 'child_process';
+import { FrontendApplicationConfig } from '@theia/application-package/lib/application-props';
+import URI from '../common/uri';
+import { FileUri } from '../node/file-uri';
+import { Deferred } from '../common/promise-util';
+import { MaybePromise } from '../common/types';
+import { ContributionProvider } from '../common/contribution-provider';
+import { ElectronSecurityToken } from '../electron-common/electron-token';
+const Storage = require('electron-store');
+const createYargs: (argv?: string[], cwd?: string) => Argv = require('yargs/yargs');
+
+/**
+ * Options passed to the main/default command handler.
+ */
+export interface MainCommandOptions {
+
+    /**
+     * By default, the first positional argument. Should be either a relative or absolute file-system path pointing to a file or a folder.
+     */
+    readonly file?: string;
+
+}
+
+/**
+ * Fields related to a launch event.
+ *
+ * This kind of event is triggered in two different contexts:
+ *  1. The app is launched for the first time, `secondInstance` is false.
+ *  2. The app is already running but user relaunches it, `secondInstance` is true.
+ */
+export interface ExecutionParams {
+    readonly secondInstance: boolean;
+    readonly argv: string[];
+    readonly cwd: string;
+}
+
+export const ElectronApplicationGlobals = Symbol('ElectronApplicationSettings');
+export interface ElectronApplicationGlobals {
+    readonly THEIA_APP_PROJECT_PATH: string
+    readonly THEIA_BACKEND_MAIN_PATH: string
+    readonly THEIA_FRONTEND_HTML_PATH: string
+}
+
+export const ElectronMainContribution = Symbol('ElectronApplicationContribution');
+export interface ElectronMainContribution {
+    /**
+     * The application is ready and is starting. This is the time to initialize
+     * services global to this process.
+     *
+     * This event is fired when the process starts for the first time.
+     */
+    onStart?(application?: ElectronApplication): MaybePromise<void>;
+    /**
+     * The application is stopping. Contributions must perform only synchronous operations.
+     */
+    onStop?(application?: ElectronApplication): void;
+}
+
+// Extracted the functionality from `yargs@15.4.0-beta.0`.
+// Based on https://github.com/yargs/yargs/blob/522b019c9a50924605986a1e6e0cb716d47bcbca/lib/process-argv.ts
+@injectable()
+export class ProcessArgv {
```

Comment:
`ElectronProcessArgv`

Reply:
I will change it. The idea was to have it here until we bump up the `yargs` version. Currently, it is in beta.

Reply:
You could try to bump it up in this PR. We don't need CQs for runtime deps. A license check from us is enough.

Reply:
It's not easy. If I bump it up in Theia, there will be other (transitive) dependencies using an older version of `yargs`, and will mess up the hosting 😕